webapp/src/main/java/life/qbic/views/login/LoginHandler.java

```diff
@@ -37,25 +42,35 @@ public void handle(LoginLayout loginView) {
       registeredLoginView = loginView;
     }
     initFields();
-    resetMessages();
     addListener();
   }
+
   private void initFields() {
-    registeredLoginView.password.setHelperText("A password must be at least 8 characters");
-    registeredLoginView.password.setPattern(".{8,}");
-    registeredLoginView.password.setErrorMessage("Invalid Password");
-    registeredLoginView.email.setErrorMessage("Invalid Email");
+    clearMessages();
+  }
+
+  private void clearMessages() {
+    registeredLoginView.clearErrors();
+    registeredLoginView.clearInformation();
+  }
+
+  private void showInvalidCredentialsError() {
+    registeredLoginView.showError(INCORRECT_USERNAME_OR_PASSWORD);
+  }
+
+  private void showEmailConfirmationInformation() {
+    registeredLoginView.showInformation(EMAIL_CONFIRMATION_SUCCESS);
   }
 
   private void addListener() {
-    registeredLoginView.loginButton.addClickShortcut(Key.ENTER);
-    registeredLoginView.loginButton.addClickListener(
-        event -> {
-          resetMessages();
-          resetComponentErrors();
-          checkUserEmail(registeredLoginView.email.getValue());
-        });
-    //ToDo Add forgot password Logic
+    registeredLoginView.addLoginListener(it -> onLoginSucceeded());
+    registeredLoginView.addForgotPasswordListener(it -> {/*TODO*/});
+    registeredLoginView.addErrorListener(() -> System.out.println("error occurred"));
+  }
```

Comment:
NIT: I tested this with invalid credentials and the listener is never triggered. 
One solution could be that the`handle` method could fire an event that triggers the listener which then sets the invalidcredentials message. 
Down the line we should also discuss if it's possible to extract the error reason from the beforeEvent parameter.

Reply:
Yes it is bypassed and I removed it now as we do not need it :+1: good spot

Reply:
The login should never give a reason! It only fails.

Reply:
I agree that it should not specify which credential is wrong but what about other kind of Exceptions (unexpected Exceptions)?

Reply:
the user should not be informed that there was an unexpected exception.

Reply:
that would expose internals

Reply:
There should still be a way to differentiate between showing an invalid credential error message (prompting the user to check his input) and an error message that something went wrong on our side